src/NuGet/NuGetProjectPackUtil.csproj

```diff
@@ -10,91 +10,103 @@
-    <NuspecProperties>$(NuspecProperties);SystemRuntimeLoaderVersion=$(SystemRuntimeLoaderVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemRuntimeInteropServicesVersion=$(SystemRuntimeInteropServicesVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemRuntimeNumericsVersion=$(SystemRuntimeNumericsVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemSecurityAccessControlVersion=$(SystemSecurityAccessControlVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemSecurityCryptographyAlgorithmsVersion=$(SystemSecurityCryptographyAlgorithmsVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemSecurityCryptographyEncodingVersion=$(SystemSecurityCryptographyEncodingVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemSecurityCryptographyX509CertificatesVersion=$(SystemSecurityCryptographyX509CertificatesVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemTextEncodingVersion=$(SystemTextEncodingVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemTextEncodingCodePagesVersion=$(SystemTextEncodingCodePagesVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemTextEncodingExtensionsVersion=$(SystemTextEncodingExtensionsVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemTextRegularExpressionsVersion=$(SystemTextRegularExpressionsVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemThreadingVersion=$(SystemThreadingVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemThreadingTasksVersion=$(SystemThreadingTasksVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemThreadingTasksExtensionsVersion=$(SystemThreadingTasksExtensionsVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemThreadingTasksParallelVersion=$(SystemThreadingTasksParallelVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemThreadingThreadVersion=$(SystemThreadingThreadVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemValueTupleVersion=$(SystemValueTupleVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemXmlReaderWriterVersion=$(SystemXmlReaderWriterVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemXmlXDocumentVersion=$(SystemXmlXDocumentVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemXmlXmlDocumentVersion=$(SystemXmlXmlDocumentVersion)</NuspecProperties>
-    <NuspecProperties>$(NuspecProperties);SystemXmlXPathXDocumentVersion=$(SystemXmlXPathXDocumentVersion)</NuspecProperties>
   </PropertyGroup>
+  
+  <Target Name="_InitializeNuspecProperties" 
+          BeforeTargets="GenerateNuspec"
+          DependsOnTargets="InitializeSourceControlInformation">
+
+    <PropertyGroup>
```

Comment:
Why did you take this approach vs. the one previously here of not using a target?

Reply:
The values of `SourceRevisionId`, `RepositoryType` and `RepositoryUrl` are not available at evaluation time since they are set by git task. Setting `NuspecProperties` needs to be delayed until these values are available, that is when `InitializeSourceControlInformation` is finished executing.